impl/src/http.test.ts

```diff
@@ -29,6 +29,9 @@ function runTests(cases: readonly TestCase[]): void {
 }
 
 runTests([
+  { name: "invalid-structured-header-syntax", input: "!" },
+  { name: "not-structured-header-dictionary", input: "histogram-index" },
```

Comment:
```suggestion
  { name: "not-structured-header-dictionary", input: "histogram-index" },
  { name: "a-different-type", input: "10" },
```

The "histogram-index" is probably the best test here, because it starts with a label, but that's subtle.